docs/migration-guides/v2/lib.md

```diff
@@ -0,0 +1,284 @@
+- `RealtimePresence`’s `on` method has been renamed to `subscribe`. Update your code to use `subscribe`.
+- `RealtimePresence`’s `off` method has been renamed to `unsubscribe`. Update your code to use `unsubscribe`.
+- `Auth`’s `authorise` method has been renamed to `authorize`. Update your code to use `authorize`.
+- The `"xhr"` transport has been renamed to `"xhr_streaming"`. Update your client options code to use `transports: ["xhr_streaming"]`.
+
+### Only TypeScript users
+
+- In v2, the `stats()` method on `Rest` and `Realtime` no longer accepts an argument of type `any`. Make sure that any argument you pass to this method implements the `StatsParams` interface.
+- In v2, `fromEncoded<T>` and `fromEncodedArray<T>` types, which were already not being used by the library, are no longer exported by the library. Remove your references to these types.
+
+<h2 id="stop-using-v2-removed">Stop using other functionality that v2 removes</h2>
+
+<h3 id="switch-from-callbacks-to-promises">Switch from the callbacks-based variant of the library to the promise-based variant</h3>
+
+ably-js v1 offered a choice between styles of asynchronous programming. There was a variant of the library that implemented asynchronous function calls via callbacks, and another that implemented them via promises. Now that promises are widely supported across modern JavaScript engines, ably-js v2 removes the callbacks variant of the library, and only offers promise-based asynchronicity.
+
+So, if you’re currently using the callbacks variant of the library, then before upgrading to v2 you should switch to using the promises variant. Here, we explain how to do this.
+
+First, you should stop choosing the callbacks variant of the library, and instead choose the promises variant. How exactly you should change your code to achieve this depends on how you’re currently choosing the callbacks variant:
+
+- If you’re implicitly choosing the callbacks variant of the library by writing `require('ably')`, then update your code to `require('ably/promises')`.
+- If you’re explicitly choosing the callbacks variant through the subpath of the imported module — that is, if you’re writing `require('ably/callbacks')` — then update your code to `require('ably/promises')`.
+- If you’re explicitly choosing the callbacks variant at the moment of instantiating the client — that is, if you’re writing `new Ably.Rest.Callbacks(…)` or `new Ably.Realtime.Callbacks(…)` — then update your code to use the `Promise` property instead, i.e. write `new Ably.Realtime.Promise(…)` or `new Ably.Rest.Promise(…)`.
+
+Now, update your code to use ably-js’s promise-based API instead of the callback-based API. **(TODO what would be good guidance on how to do this? Is it just "look at the v1 documentation for the promises variant and figure it out, here's a link", or an example of a single method call plus some hand-wavey instructions, or something else?)**
```

Comment:
Thoughts?

Reply:
Thanks for the suggestion; I've pushed some stuff along the lines you suggested.

Reply:
Awesome changes, thank you!